src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -30,7 +30,7 @@ import scala.reflect.ClassTag
   *  @define willNotTerminateInf
   */
 @SerialVersionUID(1L)
-class ArrayDeque[A] private(var array: Array[AnyRef], var start: Int, var end: Int)
+class ArrayDeque[A] private(var array: Array[Any], var start: Int, var end: Int)
```

Comment:
Be aware that this does have impact on performance.
Now the array may be unboxed but accesses to it will be intercepted and dispatched to `scala.runtime.ScalaRunTime`.
When benchmarking  this implementation warmup runs should include underlying arrays of different type to make-up realistic polluted type profile.